l10n_br_account_payment_order/models/account_move_line.py

```diff
@@ -162,11 +162,12 @@ def _prepare_payment_line_vals(self, payment_order):
 
             # Se for uma solicitação de baixa do título é preciso informar o
             # campo debit o codigo original coloca o amount_residual
-            if (
-                self.mov_instruction_code_id.id
-                == self.payment_mode_id.cnab_write_off_code_id.id
-            ):
-                vals["amount_currency"] = self.credit or self.debit
+            if self.payment_mode_id.cnab_write_off_code_id:
+                if (
+                    self.mov_instruction_code_id.id
+                    == self.payment_mode_id.cnab_write_off_code_id.id
+                ):
```

Comment:
@kaynnan experimente fazer assim, para evitar os if aninhados:

```suggestion
            if (
                self.payment_mode_id.cnab_write_off_code_id:
                and self.mov_instruction_code_id.id
                == self.payment_mode_id.cnab_write_off_code_id.id
            ):
```

Reply:
Bem melhor, valeu @antoniospneto